app/src/main/java/com/android/feedme/ui/camera/Camera.kt

```diff
@@ -216,22 +220,47 @@ fun CameraScreen(navigationActions: NavigationActions, inputViewModel: InputView
                       modifier = Modifier.testTag("PhotoSavedMessage"))
                 }
           }
-          // If text recognition button is pressed
-          if (displayText.value) {
-            cameraViewModel.textRecognitionButtonPressed()
-            OverlayTextField(
-                isVisible = true,
-                onDismiss = { displayText.value = false },
-                text = cameraViewModel.informationToDisplay.collectAsState().value)
-          }
+          // Snack bar host for info messages (green)
+          SnackbarHost(
+              hostState = snackbarHostStateInfo,
+              modifier = Modifier.align(Alignment.TopCenter),
+              snackbar = { snackbarData ->
+                Snackbar(
+                    snackbarData = snackbarData,
+                    containerColor = Color.Green.copy(alpha = 0.5f),
+                    contentColor = Color.White)
+              })
 
-          // If barcode scanner button is pressed
-          if (displayBarcode.value) {
-            cameraViewModel.barcodeScanButtonPressed()
-            OverlayTextField(
-                isVisible = true,
-                onDismiss = { displayBarcode.value = false },
-                text = cameraViewModel.informationToDisplay.collectAsState().value)
+          // Snack bar host for error messages (red)
+          SnackbarHost(
+              hostState = snackbarHostStateError,
+              modifier = Modifier.align(Alignment.TopCenter),
+              snackbar = { snackbarData ->
+                Snackbar(
+                    modifier = Modifier.testTag("Error Snack Bar"),
+                    snackbarData = snackbarData,
+                    containerColor = Color.Red.copy(alpha = 0.5f),
+                    contentColor = Color.White)
+              })
+
+          // Information snack bar is displayed each time there is something new to display from the
+          // view model
+          LaunchedEffect(Unit) {
+            cameraViewModel.informationToDisplay.collect {
+              Log.d("Snack bar", "Snack bar information")
+              if (it != null) {
+                snackbarHostStateInfo.showSnackbar(message = it, duration = SnackbarDuration.Short)
```

Comment:
Hey, good reflexion ! In fact i'm gonne keep the short SnackbarDuration (4sec), because the long one is really too long (10sec). I think 4 second is enough to be visible but not too persistent, creating user frustration.